robot/web/web.go

```diff
@@ -481,8 +481,18 @@ func (svc *webService) updateResources(resources map[resource.Name]resource.Reso
 			if ok {
 				apiResColl = reg.MakeEmptyCollection()
 			} else {
-				svc.logger.Debugw("making heterogeneous API resource collection", "api", a)
-				apiResColl = resource.NewEmptyAPIResourceCollection[resource.Resource](a)
+				// log a warning here to remind users to register their APIs, but can ignore if client communication is not wanted/needed
+				// or if resource is internal to RDK
```

Comment:
I understand `heterogeneous API resource collection` as a resource collection consisting of multiple APIs (not actually sure if the code as it was written would do that though, since the resources are grouped beforehand). this block of code is not needed since we only go into this block if the specific api registration also doesn't exist, which means we are unable to serve these resources anyway. In cases for foreign apis e.g: apis which are unknown to rdk, those requests actually goes through a different path `ForeignServiceHandler` or something like that. Foreign api resources are also exclusively remote, so we don't print the warning if all resources of an unknown api is remote. 

we actually have one service with no proto contract (base_remote_control) (and encoder used to have no proto as well), so there is precedence for APIs which don't need a rpc service or client.

Reply:
Thanks for the discussion here and offline. This change LGTM, but can we elaborate on this comment a bit? Something like (assuming this is correct):
```
// Log a warning here to remind users to register their APIs. Do not warn if the resource is internal to the RDK or
// the resource is handled by a remote with a possibly separate API registration. Modular resources will 
// have API registrations already and should not reach this point in the method.

Reply:
done